### PR TITLE
Various small build fixes

### PR DESCRIPTION
**Pull request by carstene1ns**
Still investigating the WiiU problem, but Vita and 3DS work again.

### CODE REVIEW

**Comment by Ghabry** [comment]
what appears to work for Wiiu is passing ``-Db_staticpic=false`` as an argument to meson.

Or in meson-cross.txt:

```
[built-in options]
b_staticpic = false
```

**Comment by Ghabry** [comment]
Not useful (?) because harfbuzz is disabled on WiiU but this patch fixes the build:

```patch
diff -Naur harfbuzz-8.3.0-orig/meson.build harfbuzz-8.3.0/meson.build
--- harfbuzz-8.3.0-orig/meson.build	2024-03-29 23:30:58.396713998 +0100
+++ harfbuzz-8.3.0/meson.build	2024-03-29 23:34:46.996124668 +0100
@@ -331,13 +331,6 @@
 
 # threads
 thread_dep = null_dep
-if host_machine.system() != 'windows'
-  thread_dep = dependency('threads', required: false)
-
-  if thread_dep.found()
-    conf.set('HAVE_PTHREAD', 1)
-  endif
-endif
 
 conf.set_quoted('PACKAGE_NAME', 'HarfBuzz')
 conf.set_quoted('PACKAGE_VERSION', meson.project_version())
```

File: shared/mk-meson-cross.sh [hunk 1]
```diff
@@ -123,6 +125,7 @@ c_args = [` array2list ${CPPFLAGS} ${CFLAGS} `]
 c_link_args = [` array2list ${LDFLAGS} ${LIBS} `]
 cpp_args = [` array2list ${CPPFLAGS} ${CXXFLAGS} `]
 cpp_link_args = [` array2list ${LDFLAGS} ${LIBS} `]
+${PLAT_OPTIONS_ADD}
```
Comment (Ghabry) [inline shared/mk-meson-cross.sh:128]: very pragmatic solution :D - but it works.

**Review by fdelapena — APPROVED** [review]

**Review by Ghabry — APPROVED** [review]
I like the color output :+1:

**Pull request merged**